buildlib/pr/main.yml

```diff
@@ -398,6 +398,19 @@ stages:
     jobs:
       - template: cuda/cuda.yml
 
+
+  - stage: AddressSanitizer
+    dependsOn: [Static_check]
+    jobs:
+    - template: tests.yml
+      parameters:
+        name: gpu
+        demands: ucx_gpu -equals yes
+        test_perf: 0
+        container: ubuntu22_cuda_12_1
+        asan_check: yes
+        valgrind_disable: yes
```

Comment:
maybe change valgrind_disable to valigrind_check to make it consistent

Reply:
If you don't object, I would prefer to make it in a separate PR.

Reply:
ok